test/integration/controller/jobs/job/job_controller_test.go

```diff
@@ -588,12 +588,17 @@ var _ = ginkgo.Describe("Job controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 			})
 
 			ginkgo.By("verify the PodSetUpdates are restored", func() {
-				gomega.Expect(createdJob.Spec.Template.Annotations).ShouldNot(gomega.HaveKey("ann1"))
-				gomega.Expect(createdJob.Spec.Template.Annotations).Should(gomega.HaveKeyWithValue("old-ann-key", "old-ann-value"))
-				gomega.Expect(createdJob.Spec.Template.Labels).ShouldNot(gomega.HaveKey("label1"))
-				gomega.Expect(createdJob.Spec.Template.Labels).Should(gomega.HaveKeyWithValue("old-label-key", "old-label-value"))
-				gomega.Expect(createdJob.Spec.Template.Spec.NodeSelector).ShouldNot(gomega.HaveKey(instanceKey))
-				gomega.Expect(createdJob.Spec.Template.Spec.NodeSelector).ShouldNot(gomega.HaveKey("selector1"))
+				// In case of batch/Job the stop is done with multiple API calls, suspended=true being
+				// done before the info restoration. We should retry the read if the Info is not restored.
+				gomega.Eventually(func(g gomega.Gomega) {
```

Comment:
Uhm, if we verify the startTime in L583-L588, the number of failures due to flaky might be reduced.

However, maybe the way can not completely resolve flaky test.

Reply:
It's likely that it will just make it to reproduce, the restore is done in yet another api update after that.
https://github.com/kubernetes-sigs/kueue/blob/a64cdd5eeb23a066e83a79ecdc4818e2cccdcc70/pkg/controller/jobs/job/job_controller.go#L160-L185

Reply:
Yes, that's right. So, I agree with your idea.